domain/consensus/utils/txscript/opcode.go

```diff
@@ -1068,17 +1068,16 @@ func opcodeReturn(op *parsedOpcode, vm *Engine) error {
 	return scriptError(ErrEarlyReturn, "script returned early")
 }
 
-// verifyLockTime is a helper function used to validate locktimes.
-func verifyLockTime(txLockTime, threshold, lockTime uint64, isRelativeLockTime bool) error {
-	// The lockTimes in both the script and transaction must be of the same
-	// type.
-	if !isRelativeLockTime && !((txLockTime < threshold && lockTime < threshold) ||
-		(txLockTime >= threshold && lockTime >= threshold)) {
-		str := fmt.Sprintf("mismatched locktime types -- tx locktime "+
-			"%d, stack locktime %d", txLockTime, lockTime)
-		return scriptError(ErrUnsatisfiedLockTime, str)
+func verifyLockTimeWithThreshold(txLockTime, threshold, lockTime uint64) error {
+	err := verifyLockTime(txLockTime, lockTime)
+	if err != nil {
+		return err
 	}
+	return verifyLockTimeTypes(txLockTime, threshold, lockTime)
```

Comment:
Please choose between `threshold` and `types`. I also prefer that `verifyLockTimeTypes` will happen before `verifyLockTime.`